### PR TITLE
[WIP] Ubuntu updates

### PR DESCRIPTION
**Pull request by raoulbhatia**
* Ubuntu: Add 16.04 Xenail Xerus and 16.10 Yakkety Yak.
* Ubuntu: Issue EOL warning for unsupported release.
* Tests: Misc spec cleanup

### CODE REVIEW

**Comment by raoulbhatia** [comment]
@alexjfisher I am not sure how to properly test for the warnings. Maybe you can let me know how to properly do the "acceptance" testing?

**Comment by bastelfreak** [comment]
thanks for the PR, looks good so far! :+1:

**Comment by jyaworski** [comment]
@raoulbhatia I believe you're looking for this matcher re: warning testing.

https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/output-matcher

**Comment by raoulbhatia** [comment]
@jyaworski I will not have time in the near future to implement those output matchers.
Happy if somebody would step in to help adding such a test.

**Comment by erezny** [comment]
Tested https://github.com/voxpupuli/puppet-unattended_upgrades v2.0.0 on Xenial and it seems to work.

File: manifests/params.pp [hunk 1]
```diff
@@ -63,28 +68,31 @@
...
         'precise': {
           $legacy_origin      = true
           $origins            = [
             '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-            #'${distro_id}:${distro_codename}-updates', #lint:ignore:single_quote_string_with_variables
           ]
 
         }
-        'trusty', 'utopic', 'vivid', 'wily': {
+        'trusty', 'wily': {
```
Comment (igalic) [inline manifests/params.pp]: is there a particular reason you're dropping utopic and vivid?

Comment (raoulbhatia) [inline manifests/params.pp]: @igalic Ubuntu utopic [1] and vivid [2] have reached EOL and are / will be removed from the regular mirrors and migrated to the archive repositories.

The deprecation warning that is also part of the pull request will hint the user to this fact.

Do you see valid use cases for still supporting utopic and vivid?

Raoul

[1] https://lists.ubuntu.com/archives/ubuntu-announce/2015-July/000198.html
[2] https://lists.ubuntu.com/archives/ubuntu-announce/2016-January/000203.html

**Comment by jyaworski** [comment]
@raoulbhatia can you rebase on the current master?

**Comment by raoulbhatia** [comment]
@jyaworski done

**Pull request merged**